File: LeetCode/Math/Python/Leetcode478-generate-random-point-in-a-circle.py
class Solution:

    def __init__(self, radius: float, x_center: float, y_center: float):
        self.x = x_center
        self.y = y_center
        self.r = radius

    def randPoint(self) -> List[float]:
        a,b = random.random() * 2 - 1,random.random() * 2 - 1
        if a * a + b * b > 1: return  self.randPoint()
        return [self.x + a * self.r,self.y + b * self.r]

File: LeetCode/Math/JS/Leetcode478-generate-random-point-in-a-circle.js
var Solution = function (radius, x_center, y_center) {
    [this.x, this.y, this.r] = [x_center, y_center, radius];
};

Solution.prototype.randPoint = function () {
    let [a, b] = [Math.random() * 2 - 1, Math.random() * 2 - 1];
    if (a * a + b * b > 1) return this.randPoint();
    return [this.x + a * this.r, this.y + b * this.r];
};